### ISSUE TITLE
Shared Memory Error running tutorial notebook 00b_How_to_use_numpy_arrays_in_fastai

### ISSUE THREAD

**Issue by williamsdoug** [timeseriesAI/tsai#54]
Tutorial notebook fails in cell 38  error: RuntimeError: DataLoader worker (pid 23402) is killed by signal: Bus error. It is possible that dataloader's workers are out of shared memory. Please try to raise your shared memory limit.  The above exception was the direct cause of the following exception.

Is there a recommended shared memory configuration for the current release of tsai?

Gist containing notebook with error output: https://gist.github.com/williamsdoug/dc3fd738457496aa45f34aadc0479616

tsai       : 0.2.14
fastai     : 2.2.5
fastcore   : 1.3.19
torch      : 1.7.1
cudatoolkit 10.2

OS: Ubuntu 20.04.2 LTS (Focal Fossa)
GPU: NVIDIA RTX2080ti  (driver version: 460.32.03)

**Comment by williamsdoug** [comment]
It appears that the sample notebook is requiring >16G of shared memory, possibly due to some form of memory leak.

BEFORE RUNNING NOTEBOOK:

$ df –k /dev/shm
df: –k: No such file or directory
Filesystem     1K-blocks   Used Available Use% Mounted on
tmpfs           16388240 305372  16082868   2% /dev/shm


AFTER RUNNING NOTEBOOK

$ df –k /dev/shm
df: –k: No such file or directory
Filesystem     1K-blocks     Used Available Use% Mounted on
tmpfs           16388240 16086628    301612  99% /dev/shm

**Comment by geoHeil** [comment]
Are you working on a regression task? https://github.com/timeseriesAI/tsai/issues/49 might be related

**Comment by williamsdoug** [comment]
Thanks geoHeil.  I was just running the tutorial notebooks as a form of
installation basic tsai verification.  Based on the y values in the
included  tutorial notebook, I suspect this is a classification problem
rather than regression.  Just a guess, but it smells like a memory leak
issue since it consumes >16G of shared memory.  It's not a blocker for me,
since I can operate on-disk, although it's interesting that the included
tutorial notebook can serve as a reproducer for this problem.

The good news is that the remainder of tutorial notebook 00b_xxx seems to
work, so that the problem appears to be localized to a small number of
cells.

**Comment by oguiza** [comment]
Hi @williamsdoug, 
Thanks for using `tsai`.
I've just run the nb same notebook in Google Colab and it works fine end-to-end. You can try it too.

The only differencea I can see is that Colab uses Pytorch 1.7.0+cu101 and python 3.6.9 and you are using torch : 1.7.1,
cudatoolkit 10.2 and python 3.8. 

tsai       : 0.2.14
fastai     : 2.2.5
fastcore   : 1.3.19
torch      : 1.7.0+cu101
python     : 3.6.9

I use another computer that has
torch      : 1.7.0
python     : 3.7.6

and everything works fine as well. 

But I'm not sure how to help you fix this issue. Do you have any ideas?

**Comment by williamsdoug** [comment]
Thanks so much for your quick response.  By the way, TSAI is amazing!

I built another conda environment mimicking your configuration and everything now works!  The earlier error may be some form of complex package incompatibility issue.  

Here's the command sequence I used to build the environment that works with the tutorial notebook 00b_xxx .  

$ conda create --name tsai2_ref pytorch=1.7.0 cudatoolkit=10.2 fastai=2.2.5 nb_conda python=3.7.6 -c fastai -c pytorch
$ conda activate tsai2_ref
$ pip install tsai==0.2.14

Using Python 3.8 also seems to work using the following conda environment:

$ conda create --name tsai2_ref3 pytorch=1.7.0 cudatoolkit=10.2 fastai=2.2.5 nb_conda python=3.8 -c fastai -c pytorch
$  conda activate tsai2_ref3
$  pip install tsai

I further verified that the inclusion of wfdb (physionet) and xgboost packages into the same conda environment was not responsible for the error.
